### ISSUE TITLE
Linking OneSignal.framework on Nativescript project

### ISSUE THREAD

**Issue by roblav96** [OneSignal/OneSignal-iOS-SDK#60]
# You need to add a `module.modulemap` file to your .framework bundle

# Look here: https://github.com/OneSignal/OneSignal-iOS-SDK/issues/61

I've been at this for about 6 hours now and it's driving me nuts.

I've tried [Import OneSignal into your Xcode project with CocoaPods](https://documentation.onesignal.com/docs/installing-the-onesignal-ios-sdk#1-import-onesignal-into-your-xcode-project-with-co)
I've tried [Import the framework manually](https://documentation.onesignal.com/docs/installing-the-onesignal-ios-sdk#or-import-the-framework-manually-click-to-expand)

I've tried `NOTE (React Native only) : add pod 'OneSignal', '~> 1.13.2' instead.` on the CocoaPods.

I start a blank project `tns create blank` and then I manually install a `Podfile` and follow SDK installation instructions or install `OneSignal.framework` manually.

I have no clue what else to try.

I test and see if it works, I do this:
```typescript
import * as application from 'application';
declare var OneSignal: any

if (application.ios) {
	class MyDelegate extends UIResponder implements UIApplicationDelegate {

		public static ObjCProtocols = [UIApplicationDelegate]

		private applicationDidFinishLaunchingWithOptions(app: UIApplication, launchOptions: NSDictionary): boolean {
			console.log('applicationDidFinishLaunchingWithOptions >')
			
			console.dump(OneSignal)

			return true
		}

	}
	application.ios.delegate = MyDelegate
}

application.start({ moduleName: 'main-page' });
```
But I know it's not going to work, because in the build logs I don't even see anything about `OneSignal.framework` being linked to the build.

Here's my CocoaPods `env` Stack:
```
   CocoaPods : 1.1.0.beta.1
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin15]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.11.4 (15E65)
       Xcode : 7.3.1 (7D1014)
         Git : git version 2.7.4 (Apple Git-66)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 0cf8815bad367f25d07eeda63c342d379a090198
```

When trying to use the `Podfile` method, I do get this warning:
```
[!] CocoaPods did not set the base configuration of your project because your project already has a custom config set. In order for CocoaPods integration to work at all, please either set the base configurations of the target `test` to `Pods/Target Support Files/Pods-test/Pods-test.debug.xcconfig` or include the `Pods/Target Support Files/Pods-test/Pods-test.debug.xcconfig` in your build configuration (`test/build-debug.xcconfig`).
```
`test` being the name of the project.

**Comment by roblav96** [comment]
I'm also trying to play with the `build.xcconfig` file:
```
FRAMEWORK_SEARCH_PATHS = $(inherited) "$PODS_CONFIGURATION_BUILD_DIR/OneSignal/iOS_SDK/Framework"
OTHER_CFLAGS = $(inherited) -iquote "$PODS_CONFIGURATION_BUILD_DIR/OneSignal/OneSignal.framework/Headers"
OTHER_LDFLAGS = $(inherited) -ObjC -framework SystemConfiguration -framework OneSignal
```

**Comment by roblav96** [comment]
Whenever I try to install manually it says `OneSignal.framework/Info.plist does not exist`

**Comment by roblav96** [comment]
# TLDR
You need to add a `module.modulemap` file to your .framework bundle per this post http://fluentreports.com/blog/?p=242

**Comment by JKalash** [comment]
Did you try adding a bridging header file? [Adding bridging header](https://documentation.onesignal.com/docs/installing-the-onesignal-ios-sdk#3a-swift-only-add-bridging-header-click-to-expand)

**Comment by roblav96** [comment]
@JKalash I saw that, but didn't try because I wouldn't be writing my app in Swift. Also I'm trying to make it seamlessly easy for the user to run `tns install plugin nativescript-onesignal` and be done without having to modify xocde and build config.

**Comment by JKalash** [comment]
modulemap automatically created & added to framework as of [2.1.0](https://github.com/OneSignal/OneSignal-iOS-SDK/releases/tag/2.1.0)

**Comment by roblav96** [comment]
@JKalash Thank you! I'll modify the nativescript plugin to use the cocoapods.
